framework/kms-keyring.md

```diff
@@ -266,14 +268,11 @@ in the input encrypted data key list with the following conditions, until it suc
 To attempt to decrypt a particular [encrypted data key](#structures.md#encrypted-data-key),
 OnDecrypt MUST call [KMS Decrypt](#kms-decrypt).
 
+For each [KMS Decrypt](#kms-decrypt) call, if an AWS region can be extracted from the [encrypted data key](structures.md#encrypted-data-key)'s [key provider info](structures.md#key-provider-information).
+The KMS Keyring MUST call [KMS Decrypt](#kms-decrypt) using the client supplied by the [client supplier](#client-supplier), given the region as input.
 
-If an AWS region can be extracted from the [encrypted data key](structures.md#encrypted-data-key)'s
-[key provider info](structures.md#key-provider-information), then the [KMS client](#kms-client) that calls
-[KMS Decrypt](#kms-decrypt) MUST be the client returned by the [client supplier](#client-supplier)
-when given that region as input. If an AWS region cannot be extraced from the CMK ARN, then the
-KMS Keyring MAY choose an arbitrary region.
-If the [client supplier](#client-supplier) does not provide any client for the given region for the Decrypt call,
-OnDecrypt MUST skip that particular [encrypted data key](#encrypted-data-key).
+If an AWS region cannot be extracted from the [encrypted data key](structures.md#encrypted-data-key)'s [key provider info](structures.md#key-provider-information) then the KMS Keyring MUST input to the [client supplier](#client-supplier) a value denoting an unknown region.
+If the client supplier does not provide any client for the given region for this Decrypt call, OnDecrypt MUST skip that particular [encrypted data key](#encrypted-data-key).
```

Comment:
We shouldn't specify this for OnDecrypt, as the specified behavior above will never allow us to try to obtain a region from a key that doesn't have a region. Instead, the spec at this point should state that you MUST get the region from the key. If for whatever the implementation is unable to get a region from the key at this point, something is really wrong, and the implementation should fail due to being unable to follow the specification.

Reply:
Good catch